scripts/globals/magic_utils/spell_song_enhancing.lua

```diff
@@ -0,0 +1,180 @@
+-----------------------------------
+-- Song Utilities
+-----------------------------------
+require("scripts/globals/magic_utils/parameters")
+require("scripts/globals/spell_data")
+require("scripts/globals/jobpoints")
+require("scripts/globals/status")
+require("scripts/globals/utils")
+require("scripts/globals/msg")
+-----------------------------------
+xi = xi or {}
+xi.magic_utils = xi.magic_utils or {}
+xi.magic_utils.spell_song_enhancing = xi.magic_utils.spell_song_enhancing or {}
+-----------------------------------
+-- File structure:
+-- 2 Basic Functions called by the main function.
+
+-- Table variables.
+local enhancingTable  = xi.magic_utils.parameters.enhancingSong
+
+-- Enhancing Song Potency function. (1/2)
+xi.magic_utils.spell_song_enhancing.calculateEnhancingPower = function(caster, target, spell, spellId, tier, songEffect, instrumentBoost, soulVoicePower)
+    local power       = enhancingTable[spellId][7] -- The variable we want to calculate.
+    local meritEffect = enhancingTable[spellId][5]
+    local jpEffect    = enhancingTable[spellId][6]
+    local skillCap    = enhancingTable[spellId][8]
+    local potencyCap  = enhancingTable[spellId][9]
+    local multiplier  = enhancingTable[spellId][10]
+    local divisor     = enhancingTable[spellId][11]
+
+    local singingLvl  = caster:getSkillLevel(xi.skill.SINGING) + caster:getWeaponSkillLevel(xi.slot.RANGED)
+
+    -- Get Potency bonuses from Singing Skill and Instrument Skill. TODO: Investigate JP-Wiki. Most of this makes no sense.
+    -- NOTE: Tier 1 Etudes.
+    if songEffect == xi.effect.ETUDE and tier == 1 then
+        if singingLvl >= 182 and singingLvl < 236 then
```

Comment:
There are way too many conditions here to accomplish the intended result.  I would suggest attacking this from high to low, since we only want to add if >= 182
Example:
```
if singingLvl >= 450 then
    power = power + 6
elseif singingLvl >= 397 then
    power = power + 5
...
etc
```
There's other ways to approach this as well, but with that single conditional block there's no need to check both bounds every time.  Same for the tier 2 checks